src/docs/stan-reference/examples.tex

```diff
@@ -7145,87 +7193,325 @@ \section{Fitting a Gaussian Process}\label{fit-gp.section}
+  vector[N] eta;
+}
+model {
+  vector[N] f;
+  {
+    matrix[N, N] L_Sigma = L_cov_exp_quad_ARD(x, alpha, rho, 1e-9);
+    f = L_sigma * eta;
+  }
 
-A priori, the closer $\rho_d$ is to zero, the less effect the
-predictor dimension $d$ has.  A posterior, the actual dependencies
+  alpha ~ normal(0, 1);
+  rho ~ gamma(4, 4);
+  sigma ~ cauchy(0, 1);
+  eta ~ normal(0, 1);
+
+  y ~ normal(f, sigma);
+}
+\end{stancode}
+%
+
+\subsection{Priors for Gaussian Process Parameters}
+
+Formulating priors for GP hyperparameters requires the analyst to consider the
+inherent statistical properties of a GP, the GP's purprose in the model and the
+numerical issues that may arise in Stan when estimating a GP. 
+
+Perhaps most importantly, the parameters $\rho$ and $\alpha$ are weakly
+identified because of the equivalence of multivariate Gaussians with a
+covariance matrix parameterized by a Matern kernel. The ratio of the two
+parameters is well-identified, but in practice we put independent priors on the
+two hyperparameters because these two quantities are more interpretable than
+their ratio. 
+
+\subsubsection{Priors for length-scale}
+
+GPs are a flexible class of priors, and as such, can represent a wide spectrum
+of functions. Their flexibility can lead to numerical problems when estimating 
+a model in Stan using Euclidean Hamiltonian Monte Carlo. Length-scales that are
+smaller than the scale of the input data lead to a perfect fit with the
+observed data. This can require small step sizes when exploring this region
+of the parameter space. In order to limit the amount of posterior mass that
+resides in this region, the prior for length-scale should strongly regularize
+away from length-scales that are far smaller than the scale of the input.
+
+We may wish to put further soft constraints on the length-scale, but these are
+dependent on how the GP is used in our statistical model. 
+
+If our model consists of only the GP, i.e.:
+
+\begin{align}
+  f & \sim \distro{MultiNormal}\left(0, K_{\rho,\alpha}(x)\right) \\
+  y_i & \sim \distro{Normal}(f_i, \sigma) \, \forall i \in \{1, \dots, N\} \\
+  & x \in \reals^{N \times D}
+\end{align} 
+
+we likely don't need further constraints.  We'd like to allow the GP prior to
+represent both high-frequency and low-frequency functions, so our prior should
+put non-neglible mass on both sets of functions. In this case, an inverse gamma
+will work well provided it has a left tail that puts neglible mass on
+length-scales that are small with respect to the input data.  
+
+If we're using the GP as a component in a larger model that includes an overall
+mean and fixed effects for the same variables we're using as the domain for the
+GP, i.e.:
+
+\begin{align}
+  f & \sim \distro{MultiNormal}\left(0, K_{\rho,\alpha}(x)\right) \\
+  y_i & \sim \distro{Normal}(\beta_0 + x_i \beta_{[1:D]} + f_i, \sigma) \, \forall i \in \{1, \dots, N\} \\
+  & x_i^T, \beta_{[1:D]} \in \reals{R}^D,\, x \in \reals^{N \times D}, f \in \
```

Comment:
At the end of the line, f \in ?

Also \reals{R}^D looks like fancy_R R^D when rendered